backend/DB/api/queries.py

```diff
@@ -0,0 +1,162 @@
+from sqlalchemy import func, or_, and_
+from backend.DB.api import db
+from backend.DB.api import tables
+from sqlalchemy.exc import IntegrityError
+
+
+# Queries for User
+def add_user(user):
+    try:
+        db.session.add(user)
+        db.session.commit()
+    except IntegrityError:
+        db.session.rollback()
+
+
+def get_active_users():
+    return tables.User.query.filter_by(active=True).all()
+
+
+def get_all_users():
+    return tables.User.query.all()
+
+
+def get_user_by_username(username):
+    return tables.User.query.filter_by(username=username).first()
+
+
+def delete_user(username):
+    user = tables.User.query.filter_by(username=username).first()
+    db.session.delete(user)
+    db.session.commit()
+
+
+def update_user(user):
+    tables.User.query.filter_by(username=user.username).update(
+        {'active': user.active, 'name': user.name})
+    db.session.commit()
+
+
+# Queries for pair
+def add_pair(pair):
+    try:
+        db.session.add(pair)
+        db.session.commit()
+    except IntegrityError:
+        db.session.rollback()
+
+
+def get_pair_history():
+    return tables.Pair.query.all()
+
+
+def get_pair_since_last_reward(reward_type):
+    reward_date = db.session.query(
+        db.func.max(tables.Reward.date)).filter_by(reward_type=reward_type).first()
+    if reward_date[0] is None:
```

Comment:
Does this return all pair programming events if no reward has been claimed?

Reply:
Yes, returns pair history if no rewards exist